pkg/networkmanager/wireguard.jsx

```diff
@@ -0,0 +1,210 @@
+import { ModelContext } from './model-context';
+import { useDialogs } from 'dialogs.jsx';
+
+import './wireguard.scss';
+
+const _ = cockpit.gettext;
+
+function getAddresses(addresses) {
+    return addresses.map(address => address[0] + "/" + address[1]).join(",");
+}
+
+export function WireGuardDialog({ settings, connection, dev }) {
+    const Dialogs = useDialogs();
+    const idPrefix = 'network-wireguard-settings';
+    const model = useContext(ModelContext);
+
+    const [iface, setIface] = useState(settings.connection.interface_name);
+    const [privateKey, setPrivateKey] = useState(settings.wireguard.private_key);
+    const [publicKey, setPublicKey] = useState('');
+    const [listenPort, setListenPort] = useState(settings.wireguard.listen_port);
+    const [addresses, setAddresses] = useState(getAddresses(settings.ipv4.addresses));
+    const [dialogError, setDialogError] = useState('');
+    const [peers, setPeers] = useState(settings.wireguard.peers.map(peer => ({ ...peer, allowedIps: peer.allowedIps.join(',') })));
+
+    useEffect(() => {
+        async function getPublicKey() {
+            try {
+                const key = await cockpit.script(`echo ${privateKey.trim()} | wg pubkey`, { err: 'message' });
+                setPublicKey(key.trim());
+            } catch (e) {
+                setPublicKey('');
+            }
+        }
+
+        getPublicKey();
+    }, [privateKey]);
+
+    function onSubmit() {
+        function createSettingsObj() {
+            return {
+                ...settings,
+                connection: {
+                    ...settings.connection,
+                    id: `con-${iface}`,
+                    interface_name: iface,
+                    type: 'wireguard'
+                },
+                wireguard: {
+                    private_key: privateKey.trim(),
+                    listen_port: +listenPort,
+                    peers: peers.map(peer => ({ ...peer, allowedIps: [peer.allowedIps] })),
+                },
+                ipv4: {
+                    addresses: [[addresses.split('/')[0], addresses.split('/')[1], '0.0.0.0']], // TODO: fix gateway
+                    method: 'manual',
+                    dns: [],
+                    dns_search: []
+                }
+            };
+        }
+
+        dialogSave({
+            connection,
+            dev,
+            model,
+            settings: createSettingsObj(),
+            onClose: Dialogs.close,
+            setDialogError
+        });
+    }
+
+    return (
+        <NetworkModal title={_("WireGuard settings")} onSubmit={onSubmit} dialogError={dialogError} idPrefix={idPrefix}>
+            <Name idPrefix={idPrefix} iface={iface} setIface={setIface} />
+            <FormGroup label={_('Private key')} fieldId={idPrefix + '-private-key-input'}>
+                <InputGroup>
+                    <TextInput id={idPrefix + '-private-key-input'} value={privateKey} onChange={(_, val) => setPrivateKey(val)} />
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19024-20230703-110332-55dbfd0f-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)